### DISCUSSION TITLE
`git hide -b branchname`

### DISCUSSION THREAD

**Discussion by adzenith** [arxanas/git-branchless#1278] [Ideas]
Just as we have `git move -b [commit]`, it would be great to have `git hide -b [commit]` to hide a whole subtree (and ideally delete the branch if `[commit]` is a branch). This way you don't have to figure out the root commit and then `git hide -Dr [root commit]`.

**Comment by claytonrcarter** [comment]
[Great minds think alike](https://github.com/arxanas/git-branchless/discussions/283) apparently. 😄 

> it would be great to have `git hide -b [commit]`

This is pretty easy to accomplish with revsets: `git hide 'stack(<commit>)'` should do it. 

> and ideally delete the branch if [commit] is a branch

FYI this was implemented on `master` in 1841ded20192b367a18e0dc1a4d58f5b6fcca3aa, but I don't think it's made it into a release yet. The `-D` flag was removed and "delete the branch if the commit is manually hidden" is now the default behavior.
